Updated text about setting docroot to pub. (#3575)
magento/devdocs: guides/v2.3/install-gde/tutorials/change-docroot-to-pub.md

title: Modify docroot to improve security
---

In a standard installation with an Apache web server, Magento is installed to the default web root: `/var/www/html/magento2`.
Within the `magento2` folder are:
- /pub/
- /setup/
- /var/
- other folders

 
The Magento app is served from `/var/www/html/magento2/pub`. The rest of the Magento file system is vulnerable because it is accessible from a browser. 
Setting the webroot to the `pub/` directory prevents site visitors from accessing the Web Setup Wizard and other sensitive areas of the Magento file system from a browser.

This topic describes how to change the Apache [docroot]({{ page.baseurl }}/install-gde/basics/basics_docroot.html) on an existing Magento instance to serve files from the Magento `pub/` directory, which is more secure.

{: .bs-callout .bs-callout-warning }
If you're accustomed to using the Web Setup Wizard during development, be aware that you will not be able to access it when serving files from the `pub/` directory.

## A note about nginx

> If you are using [nginx]({{ page.baseurl }}/install-gde/prereq/nginx.html) and the [`nginx.conf.sample`]({{ site.mage2200url }}nginx.conf.sample){:target="_blank"} file included in the Magento installation directory, you are probably already serving files from the `pub/` directory.

Reviewer: Don't need the blank anymore for links, just a slight edit.

```suggestion
If you are using [nginx]({{ page.baseurl }}/install-gde/prereq/nginx.html) and the [`nginx.conf.sample`]({{ site.mage2200url }}nginx.conf.sample) file included in the Magento installation directory, you are probably already serving files from the `pub/` directory.
```

Author: Thanks.

> The sample configuration overrides your server's docroot settings to serve files from Magento's `pub/` directory, assuming you have referenced the `nginx.conf.sample` in the server block that defines your site. For example, see the last line in the following configuration:

Reviewer: Just a suggested edit:

```suggestion
When used in your server block that defines your site, the `nginx.conf.sample` configuration overrides your server's docroot settings to serve files from Magento's `pub/` directory. For example, see the last line in the following configuration:
```

              set $MAGE_ROOT /var/www/html/magento2ce;
              include /var/www/html/magento2ce/nginx.conf.sample;
    }

> ```

Reviewer: Just make sure there is an empty line between the code ticks and Before you begin section.

Author: THanks.
